src/Nix/Builtins.hs

```diff
@@ -126,14 +126,15 @@ instance
   )
   => ToBuiltin t f m (a -> b) where
   toBuiltin name f =
-    pure $ mkNVBuiltin (coerce name) $ toBuiltin name . f <=< fromValue . Deeper
+    pure $ NVBuiltin (coerce name) $ toBuiltin name . f <=< fromValue . Deeper
 
 -- *** @WValue@ closure wrapper to have @Ord@
 
 -- We wrap values solely to provide an Ord instance for genericClosure
-newtype WValue t f m = WValue (NValue t f m)
+data WValue t f m where
+  WValue :: NVConstraint f => NValue t f m -> WValue t f m
```

Comment:
Used GADTs here to hide the constraint, or there is more implication?

Asking, because I know GADTs were slow in old GHCs (need to find new information on it) & here was free `newtype` before. If GADT saves only a couple of lines, it get used only here. (oh, removing the `language GADTs` extension declarations in files).

Reply:
GADTs here was just used to impose the constraint which can be impose else where.
Take performance into consideration, newtype would be better, converting it back to newtype.

Reply:
done

Reply:
Ok, thanks.